src/main/java/de/caritas/cob/messageservice/filter/HttpTenantFilter.java

```diff
@@ -0,0 +1,41 @@
+package de.caritas.cob.messageservice.filter;
+
+
+import de.caritas.cob.messageservice.api.tenant.TenantContext;
+import de.caritas.cob.messageservice.api.tenant.TenantResolver;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+
+/**
+ * Sets tenantId for current thread needed for tenant feature.
+ */
+
+@Component
+@ConditionalOnExpression("${multitenancy.enabled:true}")
+@RequiredArgsConstructor
+@Slf4j
+public class HttpTenantFilter extends OncePerRequestFilter {
+
+  private final @Nullable TenantResolver tenantResolver;
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+      FilterChain filterChain) throws ServletException, IOException {
+    log.debug("Trying to resolve tenant for request coming from URI {}", request.getRequestURI());
+    Long tenantId = tenantResolver.resolve(request);
```

Comment:
If tenantResolver is really nullable, a nullpointer exception will be raised here in case it is null.

Reply:
same answere as on other microservices. it's a leftover after refactor.